src/test/java/ca/utoronto/fitbook/unit/FollowUseCaseTest.java

```diff
@@ -0,0 +1,97 @@
+
+    private UserLocalMemoryRepository userLocalMemoryRepository;
+    private FollowUseCase followUseCase;
+    private User testUser1;
+    private User testUser2;
+
+    @BeforeAll
+    public void init() {
+        this.userLocalMemoryRepository = new UserLocalMemoryRepository();
+        this.followUseCase = new FollowService(userLocalMemoryRepository, userLocalMemoryRepository);
+
+        testUser1 = User.builder()
+                .followersIdList(new ArrayList<>())
+                .joinDate(new Date())
+                .name("Jan")
+                .totalLikes(1)
+                .password("pw")
+                .postIdList(new ArrayList<>())
+                .followingIdList(new ArrayList<>())
+                .followersIdList(new ArrayList<>())
+                .likedPostIdList(new ArrayList<>())
+                .build();
+        testUser2 = User.builder()
+                .followersIdList(new ArrayList<>())
+                .joinDate(new Date())
+                .name("Feb")
+                .totalLikes(1)
+                .password("pw2")
+                .postIdList(new ArrayList<>())
+                .followingIdList(new ArrayList<>())
+                .followersIdList(new ArrayList<>())
+                .likedPostIdList(new ArrayList<>())
+                .build();
+        userLocalMemoryRepository.save(testUser1);
+        userLocalMemoryRepository.save(testUser2);
+    }
+
+    @AfterAll
+    public void cleanUp() {
+        userLocalMemoryRepository.delete(testUser1.getId());
+        userLocalMemoryRepository.delete(testUser2.getId());
+    }
+    @AfterEach
+    public void resetFollowers(){
+        testUser1.setFollowersIdList(new ArrayList<>());
+        testUser1.setFollowingIdList(new ArrayList<>());
+        testUser2.setFollowersIdList(new ArrayList<>());
+        testUser2.setFollowingIdList(new ArrayList<>());
+        userLocalMemoryRepository.save(testUser1);
+        userLocalMemoryRepository.save(testUser2);
+    }
+
+
+    @Test
+    public void followUserReturnsValidResponse() {
+        FollowCommand followCommand = new FollowCommand(testUser1.getId(), testUser2.getId());
+
+        FollowResponse followResponse = this.followUseCase.followUser(followCommand);
+
+        Assertions.assertEquals(testUser2.getId(), followResponse.getFolloweeId());
+    }
+    @Test
+    public void followUserFollowsThemselvesThrowsUserSelfFollowingException() {
+        FollowCommand followCommand = new FollowCommand(testUser1.getId(), testUser1.getId());
+
+        Assertions.assertThrows(FollowService.UserSelfFollowingException.class, ()->this.followUseCase.followUser(followCommand));
+    }
+
+    @Test
+    public void followUserRepeatsFollowThrowsUserAlreadyFollowingException() {
+        testUser1.setFollowingIdList(List.of(testUser2.getId()));
+        testUser2.setFollowersIdList(List.of(testUser1.getId()));
+        userLocalMemoryRepository.save(testUser1);
+        userLocalMemoryRepository.save(testUser2);
```

Comment:
Modification and saving of the user should not be happening inside of tests if those users are being used by other tests, instead, make new users that are following each other.